assets/js/base/components/tabs/index.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { withInstanceId } from '@wordpress/compose';
+import classnames from 'classnames';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const TabButton = ( {
+	tabId,
+	onClick,
+	children,
+	selected,
+	ariaLabel,
+	...rest
+} ) => {
+	return (
+		<button
+			role="tab"
+			type="button"
+			tabIndex={ selected ? null : -1 }
+			aria-selected={ selected }
+			aria-label={ ariaLabel }
+			id={ tabId }
+			onClick={ onClick }
+			{ ...rest }
+		>
+			<span className="wc-component__tab-item-content">{ children }</span>
+		</button>
+	);
+};
+
+const Tabs = ( {
+	className,
+	onSelect = () => null,
+	tabs,
+	activeClass = 'is-active',
+	initialTabName,
+	instanceId,
+	ariaLabel = __( 'Tabbed Content', 'woo-gutenberg-products-block' ),
+	children,
+} ) => {
+	const [ selected, setSelected ] = useState( '' );
+	useEffect( () => {
+		if ( ! selected ) {
+			setSelected(
+				initialTabName || ( tabs.length > 0 ? tabs[ 0 ].name : '' )
+			);
+		}
+	}, [ initialTabName, tabs, selected ] );
```

Comment:
If I understand it correctly, this effect is used to set the selected tab when tabs are loaded, right? In that case, there is no need to add `selected` and `initialTabName` as dependencies because we don't need to run this when they change, is that correct?

Reply:
Ya this logic can likely be improved. I'm going to leave for #1474